### PR TITLE
support web did resolution over http for local host

### PR DESCRIPTION
**Pull request by aparkersquare**
# Overview
Uses `http` for web did resolution for local hosts. This allows for local development.

See https://github.com/TBD54566975/web5-kt/issues/324

### CODE REVIEW

File: dids/src/main/kotlin/web5/sdk/dids/methods/web/DidWeb.kt
Comment (mistermoe) [inline dids/src/main/kotlin/web5/sdk/dids/methods/web/DidWeb.kt:192]: @aparkersquare thoughts on using a parsed url to build `targetUrl`? something like

```kotlin
  public fun decodeId(parsedDid: Did): String {
    val domainNameWithPath = parsedDid.id.replace(":", "/")
    val decodedDomain = URLDecoder.decode(domainNameWithPath, UTF_8)
    val url = URL("https://$decodedDomain")

    val decodedId = StringBuilder()
    if (url.host == "localhost" || url.host == "127.0.0.1") {
      decodedId.append("http://")
    } else {
      decodedId.append("https://")
    }

    decodedId.append(decodedDomain)

    if (url.path.isEmpty()) {
      decodedId.append(WELL_KNOWN_URL_PATH)
    }
    decodedId.append(DID_DOC_FILE_NAME)

    return decodedDomain.toString()
  }
```

potential benefits of doing it this way are:
* url is parsed (therefore "validated" prior to doing anything else
* we can check `url.host` explicitly

there's probably kotlin idioms that can be used in the above suggestion to make the code more kotlin-esque which you'd know better about than me!

when i gave chad (aka chatGPT) the above suggestion and asked him for something more idiomatic, he suggested:

```kotlin
public fun decodeId(parsedDid: Did): String {
    val domainNameWithPath = parsedDid.id.replace(":", "/")
    val decodedDomain = URLDecoder.decode(domainNameWithPath, UTF_8.name())
    val url = URL("https://$decodedDomain")

    return buildString {
        append(if (url.host == "localhost" || url.host == "127.0.0.1") "http://" else "https://")
        append(decodedDomain)
        if (url.path.isEmpty()) {
            append(WELL_KNOWN_URL_PATH)
        }
        append(DID_DOC_FILE_NAME)
    }
}
```

let me know what you think! i personally prefer the former but maybe that's because i don't write much kotlin. Either way, 👍🏽

**Review by mistermoe — APPROVED** [review]
thanks @aparkersquare ! left a comment and approved regardless so you're not stuck waiting for approval

File: dids/src/main/kotlin/web5/sdk/dids/methods/web/DidWeb.kt
Comment (aparkersquare) [reply dids/src/main/kotlin/web5/sdk/dids/methods/web/DidWeb.kt:192]: Oooh, I didn't know about `url.host`!
Agree, that looks like it could be a nicer approach. Thanks.

I'll have a closer look soon.

**Review by jiyoontbd — APPROVED** [review]

File: dids/src/main/kotlin/web5/sdk/dids/methods/web/DidWeb.kt
Comment (aparkersquare) [reply dids/src/main/kotlin/web5/sdk/dids/methods/web/DidWeb.kt:192]: This works.
Updated the PR to use this approach, and deleted the LocalHostDomainMatcher

Comment (mistermoe) [reply dids/src/main/kotlin/web5/sdk/dids/methods/web/DidWeb.kt:192]: Naisuu!

**Comment by angiejones** [comment]
thanks so much @aparkersquare 🙏🏾

**Comment by taniashiba** [comment]
thank you so much!

**Pull request merged**
